experiment/measurer/measure_worker.py

```diff
@@ -78,6 +80,49 @@ def extract_corpus(corpus_archive: str, sha_blacklist: Set[str],
         filesystem.write(file_path, member_contents, 'wb')
 
 
+class StateFile:
+    """A class representing the state of measuring a particular trial on
+    particular cycle. Objects of this class are backed by files stored in the
+    bucket."""
+
+    def __init__(self, name: str, state_dir: str, cycle: int):
+        self.name = name
+        self.state_dir = state_dir
+        self.cycle = cycle
+        self._prev_state = None
+
+    def _get_gcs_cycle_state_file_path(self, cycle: int) -> str:
+        state_file_name = experiment_utils.get_cycle_file_name(
+            self.name, cycle, '.json')
+        state_file_path = os.path.join(self.state_dir, state_file_name)
+        return exp_path.gcs(pathlib.Path(state_file_path))
+
+    def _get_previous_cycle_state(self, first_cycle_default) -> str:
+        if self.cycle == 1:
+            return first_cycle_default
+
+        previous_state_file_gcs_path = self._get_gcs_cycle_state_file_path(
+            self.cycle - 1)
+        return json.loads(
+            gsutil.cat(previous_state_file_gcs_path, expect_zero=False))
+
+    def get_previous(self, first_cycle_default=None):
+        """Returns the previous state."""
+        if self._prev_state is None:
+            self._prev_state = self._get_previous_cycle_state(
+                first_cycle_default=first_cycle_default)
+
+        return self._prev_state
+
+    def set_current(self, state):
+        """Sets the state for this cycle on GCS."""
```

Comment:
consider referring to the directory instead, ie: "on GCS" -> "in the experiment data directory" (or something like that)

Reply:
Changed to "bucket"